Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSStitch1D.py

```diff
@@ -0,0 +1,421 @@
+            raise RuntimeError("The min value of the FRONT detector data set is larger"
+                               "than the max value of the REAR detector data set")
+
+        # Get the min and max range
+        min_q = max(rear_q_min, front_q_min)
+        max_q = min(rear_q_max, front_q_max)
+
+        return min_q, max_q
+
+    def _determine_factors(self, q_high_angle, q_low_angle, mode, scale, shift):
+
+        # We need to make suret that the fitting only occurs in the y direction
+        constant_x_shift_and_scale = ', f0.Shift=0.0, f0.XScaling=1.0'
+
+        # Determine the StartQ and EndQ values
+        q_min, q_max = self._get_start_q_and_end_q_values(rear_data=q_low_angle, front_data=q_high_angle,
+                                                          )
+
+        # We need to transfer the errors from the front data to the rear data, as we are using the the front data as a model, but
+        # we want to take into account the errors of both workspaces.
+        front_data_corrected, rear_data_corrected = self._get_error_corrected(rear_data=q_low_angle,
+                                                                                                       front_data=q_high_angle,
+                                                                                                       q_min=q_min, q_max=q_max)
+
+        fit = self.createChildAlgorithm('Fit')
+
+        # We currently have to put the front_data into the ADS so that the TabulatedFunction has access to it
+        front_data_corrected = AnalysisDataService.addOrReplace('front_data_corrected', front_data_corrected)
+        front_in_ads = AnalysisDataService.retrieve('front_data_corrected')
+
+        function = 'name=TabulatedFunction, Workspace="' + str(
+            front_in_ads.name()) + '"' + ";name=FlatBackground"
+
+        fit.setProperty('Function', function)
+        fit.setProperty('InputWorkspace', rear_data_corrected)
+
+        constant_x_shift_and_scale = 'f0.Shift=0.0, f0.XScaling=1.0'
+        if mode == Mode.BothFit:
+            fit.setProperty('Ties', constant_x_shift_and_scale)
+        elif mode == Mode.ShiftOnly:
+            fit.setProperty('Ties', 'f0.Scaling=' + str(scale) + ',' + constant_x_shift_and_scale)
+        elif mode == Mode.ScaleOnly:
+            fit.setProperty('Ties', 'f1.A0=' + str(shift) + '*f0.Scaling,' + constant_x_shift_and_scale)
+        else:
+            raise RuntimeError('Unknown fitting mode requested.')
+
+
+        fit.setProperty('StartX', q_min)
+        fit.setProperty('EndX', q_max)
+        fit.setProperty('CreateOutput', True)
+        fit.execute()
+        param = fit.getProperty('OutputParameters').value
+        AnalysisDataService.remove(front_in_ads.name())
+
+        # The outparameters are:
+        # 1. Scaling in y direction
+        # 2. Shift in x direction
+        # 3. Scaling in x direction
+        # 4. Shift in y direction
+        # 5. Chi^2 value
```

Comment:
output 5 is the cost function, that was a previous mistake on the ISIS SANS side